radixdlt-core/radixdlt/src/test/java/com/radixdlt/tree/substate/BerkeleySubStateStoreSpy.java

```diff
@@ -84,7 +84,11 @@ public void process(
       Function<SystemMapKey, Optional<RawSubstateBytes>> mapper) {
     super.process(dbTxn, txn, stateVersion, mapper);
     for (REStateUpdate su : txn.stateUpdates().toList()) {
-      this.reStateUpdateList.put(su.getId(), su);
```

Comment:
Perhaps it is worth to consider turning this code into a stream processing chain. This would eliminate the need to "materialize" the list.

Reply:
This is used in a test. The goal is to create the map which in the end has only the current up substates. Could please elaborate more?

Reply:
Something like this:

```
txn.stateUpdates().forEach(su -> {
  if (su.isBootUp()) {
    this.reStateUpdateList.put(su.getId(), su);
  } else {
    this.reStateUpdateList.remove(su.getId());
  });
```

Reply:
Ah I see, you mean this `txn.stateUpdates().toList()`?

Reply:
Yes